src/workspacer.FSharpConfig/FSharpCompiler.cs

```diff
@@ -0,0 +1,94 @@
+﻿namespace workspacer.FSharpConfig
+{
+    using FSharp.Compiler.CodeAnalysis;
+    using Microsoft.FSharp.Control;
+    using Microsoft.FSharp.Core;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Reflection;
+
+    public static class FSharpCompiler
+    {
+        private static FSharpChecker checker =
+            FSharpChecker.Create(
+                FSharpOption<int>.None,
+                FSharpOption<bool>.None,
+                FSharpOption<bool>.None,
+                FSharpOption<LegacyReferenceResolver>.None,
+                FSharpOption<FSharpFunc<Tuple<string, DateTime>, FSharpOption<Tuple<object, IntPtr, int>>>>.None,
+                FSharpOption<bool>.None,
+                FSharpOption<bool>.None,
+                FSharpOption<bool>.None,
+                FSharpOption<bool>.None);
+
+        private static Regex requireRegex = new Regex("^#r @\"(.*)\"");
+
+        public static void CompileScript(string scriptFile, IList<string> references, string dllFile)
+        {
+            var compilerArgs = new List<string>
+            {
+                "fsc.exe",
+                "-a",
+                scriptFile,
+                "-o",
+                dllFile,
+                $"--lib:\"{Path.GetDirectoryName(Assembly.GetCallingAssembly().Location)}\"",
+            };
+
+            foreach (var r in references.Select(Path.GetDirectoryName).Distinct())
+            {
+                compilerArgs.Add($"--lib:\"{r}\"");
+            }
+
+            var result =
+                FSharpAsync.RunSynchronously(
+                    checker.Compile(compilerArgs.ToArray(), FSharpOption<string>.None),
+                    FSharpOption<int>.None,
+                    FSharpOption<CancellationToken>.None
+                );
+            if (result.Item2 != 0)
+            {
+                throw new FSharpCompileException(result.Item1[0].Message, result.Item1);
```

Comment:
Here we seem to imply all errors when running the F# config are compilation errors.

How do we deal with runtime errors? Are they handled automagically via .NET interop?

Reply:
Then I guess this OK as is.